src/components/LocaleSwitcher.astro

```diff
@@ -0,0 +1,27 @@
+---
+import Select from 'fulldev-ui/components/Select.astro'
+import fulldevConfig from 'virtual:fulldev-ui/config'
+---
+
+{
+  Astro.props.locales && (
+    <Select
+      name="locale-switcher"
+      options={Astro.props.locales}
+      value={Astro.props.lang as string}
+      placeholder={Astro.props.lang as string}
+      radius={fulldevConfig.radius ?? 'auto'}
+    />
+  )
+}
+
+<script>
+  const select = document.querySelector('.theme-switcher select')
+  select?.addEventListener('change', (event) => {
+    const target = event.target as HTMLSelectElement
```

Comment:
Neem als regel aan om nooit 'as' te gebruiken, hiermee overwrite je als het ware de typescript errors in je IDE, maar in runtime kunnen de errors nogsteeds voorkomen.

Ik weet dat het op meer plekken wordt gebruikt, maar vgm zijn dat experimental components. In production mag het niet. 

Je kunt in plaats daarvan dit doen:

```
  select?.addEventListener('change', (event) => {
    if (event.target instanceof HTMLSelectElement) {
      const target = event.target
      const selectedLocale = target.value
      const currentUrl = new URL(window.location.href)
      currentUrl.pathname = '/' + selectedLocale + currentUrl.pathname
      window.location.href = currentUrl.toString()
    }
  })
  ```
  
  Dus een if statement ipv een as